### ISSUE TITLE
[Arcade v1] Remove stable version restriction from non-isolated feeds

### ISSUE THREAD

**Issue by brettfo** [dotnet/arcade#5487]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

## Description

PR #4682 removed the stable package version restriction in non-isolated feeds in Arcade v5.  The `dotnet/interactive` repo asks that this change be back-ported to Arcade v1 and the restriction lifted.  Currently the publish steps on our build are [failing](https://dev.azure.com/dnceng/internal/_build/results?buildId=646924&view=logs&j=51c75269-d108-5b35-500b-cc803a6d6b66&t=db467048-79f0-5bc7-fe0b-67cface4c44b).

## Customer Impact

`dotnet/interactive` publishes a dotnet global tool and we need a stable version number without a `*-beta.*` infix that is publicly available.  We previously used a custom MyGet feed for this, but internal guidance was to decomission MyGet feeds in favor of dnceng ADO feeds.

## Regression

No

## Risk

Likely low, given that the change was approved for Arcade v5 and has been functioning without issue for 4 months.

## Workarounds

`dotnet/interactive` can switch back to publishing packages via MyGet.

**Comment by JohnTortugo** [comment]
@brettfo - We are doing some re-work on the publishing infra so that we can decouple the publishing part from the .net / Arcade versioning scheme. When that work is complete you'll be able to use the publishing features that we deployed in Arcade 5.x in your branch that uses Arcade 3.x. [ETA for the feature is ~2 weeks.](https://github.com/dotnet/arcade/issues?q=is%3Aissue+is%3Aopen+post-build+improv)

/cc @mmitche

**Comment by riarenas** [comment]
@johntortugo I don't think the new publishing will be ready for the release/3.x branch for a while. Should we just port the requested change? What do you think?

**Comment by JohnTortugo** [comment]
@riarenas - I think we might be able to port the V3 publishing to servicing. Most of the changes are just YAML addition and not risky. We'll need to carefully analyze that when we merge the V3 yaml changes PR that I'll create today.

**Comment by riarenas** [comment]
I think we should port the stable check relaxation to give immediate relief while we do the work to use v3 publishing in servicing branches.

**Comment by riarenas** [comment]
This has been fixed as of #4682

**Comment by riarenas** [comment]
This still requires a version update for the task.

**Comment by riarenas** [comment]
PR out at https://github.com/dotnet/arcade/pull/5809

**Comment by riarenas** [comment]
This should now be fixed for release/3.x arcade.

**Comment by brettfo** [comment]
The dotnet/interactive repo just got the arcade update, but publishing builds to the channel fails with:

```
##[error].packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.20405.7\tools\SdkTasks\PublishArtifactsInManifest.proj(65,5): error : Method not found: 'System.Threading.Tasks.Task`1<Boolean> Sleet.PackageIndexFile.Exists(NuGet.Packaging.Core.PackageIdentity)'.
```

Build can be found [here](https://dev.azure.com/dnceng/internal/_build/results?buildId=764208&view=results).

**Comment by riarenas** [comment]
That's... really weird. It looks like the new version of the task is incompatible with newer versions of the dotnet sdk. I'll open a separate issue about it, and I'll have to revert to the previous version of the task that doesn't have this change in it.

**Comment by riarenas** [comment]
This is blocked on https://github.com/dotnet/arcade/issues/5935

**Comment by riarenas** [comment]
We finally unblocked #5935, so I'm updating this task again to a version with the restrictions removed in https://github.com/dotnet/arcade/pull/6178
